ipatests/test_integration/test_trust.py

```diff
@@ -223,6 +223,50 @@ def test_user_without_posix_attributes_not_visible(self):
         # Getent exits with 2 for non-existent user
         assert result.returncode == 2
 
+    def test_override_homedir(self):
+        """POSIX attributes should not be overwritten or missing.
+
+        Regression test for bug https://pagure.io/SSSD/sssd/issue/2474
+
+        When there is IPA-AD trust with POSIX attributes,
+        including the home directory set in the AD LDAP and in sssd.conf
+        subdomain_homedir = %o is added after initgroup call home directory
+        should be correct and do not report in logs like,
+        'get_subdomain_homedir_of_user failed: * [Home directory is NULL]'
+        """
+        tasks.backup_file(self.master, paths.SSSD_CONF)
+        log_file = "/var/log/sssd/sssd_%s.log" % self.master.domain.name
+        sssd_debug_log1 = self.master.get_file_contents(
+            log_file, encoding='utf-8')
+        assert 'get_subdomain_homedir_of_user failed' not in sssd_debug_log1
```

Comment:
The test should not fail if this string is found in sssd.log BEFORE the test. It can be there due to previous tests or any other events, not related to the test.
As I already proposed, instead you should check only part of the log which is written during test execution.